lib/importer/nunjucks/importer/macros/sheet_selector.njk

```diff
@@ -1,53 +1,114 @@
 
 {#
-    importerSheetSelector generates a radio button list which allows the 
-    user to choose which sheet is to be used to upload data. If there is 
+    importerSheetSelector generates a radio button list which allows the
+    user to choose which sheet is to be used to upload data. If there is
     only a single sheet then it is selected by default.
 
-    The list of sheet names is retrieved from the current spreadsheet 
+    The list of sheet names is retrieved from the current spreadsheet
     being uploaded, and will always contain at least one sheet.
 
-    It accepts a data object which is taken from the prototype kit's 
-    session data which is made available on every page, and contains the 
-    data submitted from forms to the backend, and also the current 
+    It accepts a data object which is taken from the prototype kit's
+    session data which is made available on every page, and contains the
+    data submitted from forms to the backend, and also the current
     data import session.
 
-    legend is the text that should be used for the legend part of the 
+    legend is the text that should be used for the legend part of the
     radio buttons, if none is supplied, then a legend is not added.
 #}
+{% from "importer/macros/table_view.njk" import importerTableView %}
+
 {% macro importerSheetSelector(data, legend) %}
   {% set selectedSheet = data['importer.session'].sheet %}
-  {% set sheets = data['importer.session'].sheets %}
+  {% set sheets = importSheetPreview(data) %}
+  {% set tableRowIndex = sheets.length + 2 %}
   {% set error = importerError(data) %}
 
-  <div class="govuk-form-group {% if error %}govuk-form-group--error{% endif %}">
-    <fieldset class="govuk-fieldset" aria-describedby="upload-error">
-    
-    {% if legend %}
-    <legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
-      <h1 class="govuk-fieldset__heading">
-        {{ legend }}
-      </h1>
-    </legend>
-    {% endif %}
-    
-    {% if error %}
-    <p id="upload-error" class="govuk-error-message">
-      <span class="govuk-visually-hidden">Error:</span> {{ error.text }}
-    </p>
-    {% endif %}
-    
-    <div class="govuk-radios" data-module="govuk-radios">
+  <style>
+  .rd-sheet-preview {
+    width: 100%;
+    display: grid;
+    grid-template-columns: 44px 1fr;
+  }
+
+  .rd-sheet-preview:not(input,label,div.selectable) {
+    display: contents;
+  }
+
+  div.selectable { display: none; overflow-x: scroll}
```

Comment:
I think you want [`overflow-x: auto`](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-x#values) else a disabled scrollbar is always shown. The default is `hidden` not `auto`.